src/controllers/project.controller.ts

```diff
@@ -185,10 +187,39 @@ export class ProjectController {
 		});
 	}
 
-	async selectDeployment(deploymentId: string): Promise<void> {
-		this.selectedDeploymentId = deploymentId;
+	setSessionsStateFilter(filterState: string) {
+		this.filterSessionsState = filterState;
+		this.fetchSessions();
+	}
 
-		const { data: sessions, error } = await SessionsService.listByDeploymentId(deploymentId);
+	async fetchSessions() {
+		if (!this.selectedDeploymentId) {
+			return;
+		}
+
+		let selectedSessionStateFilter;
+		try {
+			selectedSessionStateFilter = reverseSessionStateConverter(this.filterSessionsState as SessionStateType);
+		} catch (error) {
+			if (!this.hasDisplayedError.get("sessionConverter")) {
+				commands.executeCommand(
+					vsCommands.showErrorMessage,
+					translate().t("errors.sessionStateFilterConversionErrorExtended", { stateType: this.filterSessionsState })
+				);
```

Comment:
this is an internal error which user should report, it's not really valid user facing error
it should never happen or we should default to all if it is invalid and log the error
not sure, ask haim